outputs/kafka.go

```diff
@@ -3,31 +3,119 @@ package outputs
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"log"
+	"net"
+	"strings"
+	"time"
 
 	"github.com/DataDog/datadog-go/statsd"
 	"github.com/segmentio/kafka-go"
+	"github.com/segmentio/kafka-go/sasl/plain"
+	"github.com/segmentio/kafka-go/sasl/scram"
 
 	"github.com/falcosecurity/falcosidekick/types"
 )
 
 // NewKafkaClient returns a new output.Client for accessing the Apache Kafka.
 func NewKafkaClient(config *types.Configuration, stats *types.Statistics, promStats *types.PromStatistics, statsdClient, dogstatsdClient *statsd.Client) (*Client, error) {
 
+	transport := &kafka.Transport{
+		Dial: (&net.Dialer{
+			Timeout:   3 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		ClientID: config.Kafka.ClientID,
+	}
+
+	if config.Kafka.SASL != "" {
+		saslMode := strings.ToUpper(config.Kafka.SASL)
+		switch {
+		case saslMode == "PLAIN":
+			transport.SASL = plain.Mechanism{
+				Username: config.Kafka.Username,
+				Password: config.Kafka.Password,
+			}
+		case strings.HasPrefix(saslMode, "SCRAM_"):
+			algo := strings.TrimPrefix(config.Kafka.SASL, "SCRAM_")
+			var err error
+			switch algo {
+			case "SHA256":
+				transport.SASL, err = scram.Mechanism(scram.SHA256, config.Kafka.Username, config.Kafka.Password)
+			case "SHA512":
+				transport.SASL, err = scram.Mechanism(scram.SHA512, config.Kafka.Username, config.Kafka.Password)
+			default:
+				return nil, fmt.Errorf("unsupported SASL SCRAM algorithm %q", algo)
+			}
+			if err != nil {
+				return nil, fmt.Errorf("failed to initialize SASL SCRAM %q: %w", algo, err)
+			}
+		default:
+			return nil, fmt.Errorf("unsupported SASL mode: %q", config.Kafka.SASL)
```

Comment:
Can you also print a log line before each error:
`log.Printf("[ERROR] : Kafka - %v\n", errors.New("")` (example)

Reply:
up